rslib/src/scheduler/answering/mod.rs

```diff
@@ -363,14 +371,22 @@ impl Collection {
             }
         }
 
-        self.update_queues_after_answering_card(
-            &card,
-            timing,
-            matches!(
-                answer.new_state,
-                CardState::Filtered(FilteredState::Preview(PreviewState { finished: true, .. }))
-            ),
-        )
+        // Handle queue updates based on from_queue flag
+        if from_queue {
+            self.update_queues_after_answering_card(
+                &card,
+                timing,
+                matches!(
+                    answer.new_state,
+                    CardState::Filtered(FilteredState::Preview(PreviewState {
+                        finished: true,
+                        ..
+                    }))
+                ),
+            )
+        } else {
+            self.update_queues_after_grading_card(&card, timing)
```

Comment:
This is not very efficient - removing a single card requires a search through the queues, and the user may have selected hundreds or thousands of cards at a time. Especially for larger selections, this is probably going to perform a lot better if the seen card ids are collected into a HashSet, and then the queues are filtered with .retain() after all cards have been processed.

Alternatively we could discards the queues and have them rebuilt, which is what is currently done when using SDD.

Reply:
I implemented the first suggestion in 91e1d6f.